db/version_set.cc

```diff
@@ -1806,7 +1808,7 @@ void Version::Get(const ReadOptions& read_options, const LookupKey& k,
         cfd_->internal_stats()->GetFileReadHist(fp.GetHitFileLevel()),
         IsFilterSkipped(static_cast<int>(fp.GetHitFileLevel()),
                         fp.IsHitFileLastInLevel()),
-        fp.GetHitFileLevel());
+        fp.GetHitFileLevel(), MaxFileSizeForL0MetaPin(mutable_cf_options_));
```

Comment:
This is called in every Get(). I'm a little bit worried the overhead. Although 1-2 divisions and 1-2 multiplications are not large, but doing it for every Get() still feels a lot, while most of the time it is not needed. As a compromise, can we skip it for max_open_files=-1? Similarly for iterators too.

Reply:
Oh yeah, that's a good point... I assume the `port::kMaxSizet / 3` is a const and the `cf_options.write_buffer_size / 2` is a bit shift so hopefully there's no actual division in the Get path.. will double check

Reply:
Yes it's a good point. Maybe it's not so expensive. Also, I think it might be OK to cache the value in Version. It might prevent a change of write_buffer_size to immediately take effect, but it might be OK.

Reply:
It compiles down to one `shr` (for the second div by two) and one `lea` (for the second mult by three). The first division/multiplication can be elided as the value to compare against is precomputed.

Reply:
OK. We can leave it as it is then.

Reply:
I changed it anyways to cache in `Version`. Could be useful in the future if someone changes the ratio to 4/3 or something.